### DISCUSSION TITLE
How do use minted in latexpdf output

### DISCUSSION THREAD

**Discussion by akhmerov** [jupyter-book/mystmd#537] [Q&A] [answered]
I would like to include `{code-block} python` and `{code-cell} python` directives using `minted` package. While it seems that using
```
:::{code-block} python
:class: minted
def f(): pass
:::
```
produces the desired output (as implemented in e530b08261fe955cb351a299dc47278e68c8986a), this doesn't work with `{code-cell}` directives, and it doesn't seem possible to make this behavior a default.

Therefore I have the following questions:
- Do I understand the situation right now correctly?
- Would it be desirable to support syntax highlighting for code cells and implement this behavior globally?
- Could rendering of nodes be deferred to the templates so that it can be overridden?

Thanks in advance!

**Comment by rowanc1** [comment, marked as answer]
Hi @akhmerov -- thanks for the question. Yes, that is the current situation and I like both of your suggestions! The current approach was a bit of a stop-gap while we figure out some better ways to deal with this more generally.

We are having a bit of discussion on a global plugin options here at the moment:
https://github.com/executablebooks/mystmd/pull/536#issuecomment-1670429154

I would love any of your thinking on the best way to start turning these ideas into implementation! Also, we could fix up the `code-cell` class if that is something you need soon.

**Reply by akhmerov** [reply]
Thanks! The suggestion for using templates originates from the implementation of `nbconvert` which lets jinja2 and template inheritance do the heavy lifting.

We're trying to figure out how to use mystmd to write a paper that introduces a software package, so globally configuring code rendering and properly rendering code cells would really help!

**Comment by akhmerov** [comment]
As a quick hack that enabled us to move forward, we modified the template to contain
```
[- CONTENT
| replace("\\begin{verbatim}", "\\begin{minted}{ipython}")
| replace("\\end{verbatim}", "\\end{minted}")
| replace("\\texttt", "\\mintinline{python}") -]
```

**Reply by rowanc1** [reply]
That is really cool. Would not have thought to do it there. 🚀

Glad that you have a way forward in the short term, we will continue to work towards a better solution for this going forward.
